### PR TITLE
Global patch for t.compile high warmup time on MoE models

### PR DESCRIPTION
**Pull request by tvoas**
Removes model specific override forward() to initialize residual as zeros instead of None which eliminating torch._dynamo type guards. This is now applied Globally for all applicable MoE models.

### CODE REVIEW

File: vllm_gaudi/models/llama4.py [hunk 1]
```diff
@@ -92,13 +44,12 @@ def __init__(self, *, vllm_config: VllmConfig, prefix: str = ""):
 def _apply_hpu_llama4_init_patches(model_root: nn.Module) -> None:
     """Shared init-time patches for both CausalLM and ConditionalGeneration.
 
-    Swaps the inner model class for the residual=zeros fix and applies
-    attention type unification in compile mode.
+    Applies attention type unification in compile mode.
+    The residual=zeros warmup workaround is installed by the global MoE
+    warmup patch in HpuModelAdapter.
```
[the remark this answers is not in the document]
Comment (tvoas) [reply vllm_gaudi/models/llama4.py:49]: @copilot apply changes based on [this feedback](https://github.com/vllm-project/vllm-gaudi/pull/1452#discussion_r3254801958)

File: vllm_gaudi/models/__init__.py [hunk 2]
```diff
@@ -41,9 +41,6 @@ def register_model():
     from vllm_gaudi.models.seed_oss import HpuSeedOssForCausalLM  # noqa: F401
     ModelRegistry.register_model("SeedOssForCausalLM", "vllm_gaudi.models.seed_oss:HpuSeedOssForCausalLM")
```
[the remark this answers is not in the document]
Comment (tvoas) [reply vllm_gaudi/models/__init__.py:42]: @copilot apply changes based on [this feedback](https://github.com/vllm-project/vllm-gaudi/pull/1452#discussion_r3254801996)

File: vllm_gaudi/v1/worker/hpu_model_runner.py [hunk 3]
```diff
@@ -795,6 +797,153 @@ def _mark_unbacked_dim0(tensor: torch.Tensor):
...
+    except (OSError, TypeError):
+        return False
+    return 'residual is None' in source
+
+
+def _get_moe_warmup_patch_skip_reason(language_model: torch.nn.Module) -> str | None:
+    if not _has_moe_layers(language_model):
+        return 'no FusedMoE layers found'
+    if not hasattr(language_model, 'embed_input_ids') or not hasattr(language_model, 'norm'):
+        return 'missing embed_input_ids or norm'
```
[the remark this answers is not in the document]
Comment (tvoas) [reply vllm_gaudi/v1/worker/hpu_model_runner.py:890]: @copilot apply changes based on [this feedback](https://github.com/vllm-project/vllm-gaudi/pull/1452#discussion_r3254801968)

File: vllm_gaudi/v1/worker/hpu_model_runner.py [hunk 4]
```diff
@@ -795,6 +797,153 @@ def _mark_unbacked_dim0(tensor: torch.Tensor):
...
+        return layer
+    return None
+
+
+def _decoder_layer_uses_none_residual_sentinel(layer: torch.nn.Module) -> bool:
+    try:
+        source = inspect.getsource(layer.forward)
+    except (OSError, TypeError):
+        return False
+    return 'residual is None' in source
```
[the remark this answers is not in the document]
Comment (tvoas) [reply vllm_gaudi/v1/worker/hpu_model_runner.py:883]: @copilot apply changes based on [this feedback](https://github.com/vllm-project/vllm-gaudi/pull/1452#discussion_r3254801973)

File: vllm_gaudi/v1/worker/hpu_model_runner.py [hunk 5]
```diff
@@ -795,6 +797,153 @@ def _mark_unbacked_dim0(tensor: torch.Tensor):
...
+    # Walk through those wrappers until we reach the layer-owning module.
+    while True:
+        inner_model = getattr(language_model, 'model', None)
+        if not isinstance(inner_model, torch.nn.Module):
+            return language_model
+
+        if hasattr(language_model, 'embed_input_ids') and hasattr(language_model, 'norm') and hasattr(language_model, 'layers'):
+            return language_model
+
+        language_model = inner_model
```
[the remark this answers is not in the document]
Comment (tvoas) [reply vllm_gaudi/v1/worker/hpu_model_runner.py:859]: @copilot apply changes based on [this feedback](https://github.com/vllm-project/vllm-gaudi/pull/1452#discussion_r3254801984)
